grakn-graph/src/main/java/ai/grakn/graph/internal/RoleImpl.java

```diff
@@ -130,10 +127,11 @@ void deleteCachedDirectPlaysByType(Type oldType){
      * @return Get all the roleplayers of this role type
      */
     public Stream<Casting> rolePlayers(){
-        //TODO: This should not call reify() it should call reified() but we need a way of getting the roleplayers from relation edges which are not linked to the relation type
         return relationTypes().stream().
                 flatMap(relationType -> relationType.instances().stream()).
-                flatMap(relation -> ((RelationImpl)relation).reify().castingsRelation(this));
+                map(relation -> RelationImpl.from(relation).reified()).
+                filter(Optional::isPresent).
```

Comment:
I was initally excited about this but it's not as nice as I thought it would be :( just avoids the get call which is something I suppose